### DISCUSSION TITLE
Passing options to `has_many` serializers

### DISCUSSION THREAD

**Discussion by noxasch** [ElMassimo/types_from_serializers#21] [Q&A] [answered]
I'm having trouble when I need to pass an argument to `has_many` serializer.

Expected result:

```ts
// Items.ts

import type Price from '../Price'

export default interface Items {
  id: number
  name: string
  prices: Price[]
}
```

eg:
Let say we have an `PriceSerializer` and `ItemSerializer`. Item can have many prices. To render prrices i need to pass an option  so the serializer can accept a currency which is dynamic based on the  arguments.

Price serializer
```rb
# PriceSerializer.rb

attributes :amount

def amount
   "#{options[:currency_code]} object.amount"
end
```

I can call the serializer as this in controller

```rb
PriceSerializer.render(price, currency: 'USD')
```

by using has_many, I can't pass an argument
```rb
# ItemSerializer.rb
attributes :name

has_many :prices, serializer: PriceSerializer # doesn't accept options
```

So to overcome this

```rb
# ItemSerializer.rb
attributes :name

attribute
# PriceSerializer[] won't work since its not an actual class
type PriceSerializer 
def prices
  PriceSerializermany(object.prices, currency_code: options[:currency_code])
end
```

However in the resultant types:
```ts
// Items.ts

import type Price from '../Price'

export default interface Items {
  id: number
  name: string
  prices: Price // should be Price[]
}
```

**Comment by ElMassimo** [comment, marked as answer]
Hi Alexandr!

Passing `options` is a missing feature in `oj_serializers` that I'd like to add at some point (if I manage to do so without introducing performance regressions for the cases where `options` are not used).

In the meantime, you could try something like:

```ruby
type "Price[]"
def prices
  PriceSerializer.many(object.prices, options)
end
```

And then create an `app/frontend/types/Price.ts` file where you re-export the generated type as a [custom type](https://github.com/ElMassimo/types_from_serializers#custom_types_dir):

```ts
import type { Price } from '~/types/serializers'

export type { Price as default }
```
